modules/net-cloudnat/variables.tf

```diff
@@ -20,10 +20,25 @@ variable "addresses" {
   default     = []
 }
 
-variable "config_min_ports_per_vm" {
-  description = "Minimum number of ports allocated to a VM from this NAT config."
-  type        = number
-  default     = 64
+variable "port_allocation_config" {
+  description = "Configuration for how to assign ports to virtual machines. min_ports_per_vm and max_ports_per_vm have no effect unless enable_dynamic_port_allocation is set to 'true.'"
+  type = object({
+    enable_endpoint_independent_mapping = bool
+    enable_dynamic_port_allocation      = bool
+    min_ports_per_vm                    = number
+    max_ports_per_vm                    = number
+  })
+  default = {
+    enable_endpoint_independent_mapping = true
+    enable_dynamic_port_allocation      = false
+    min_ports_per_vm                    = 64
```

Comment:
I'm not sure if we want to keep this minimum, as the provider will default to 32? 

I've kept it here for the sake of not boiling the ocean, but setting this (and `max_ports_per_vm`) to null would allow us to validate that these are not being set when `enable_dynamic_port_allocation` is false.